### ISSUE TITLE
Why not if err != nil ?

### ISSUE THREAD

**Issue by embano1** [jdob/visitors-operator#4]
Wouldn't it be sufficient (and reduce the heap allocations due to pointers on reconcile.Result returns in the functions) to just check for `err != nil`? If there's an error return, just return both variables.

https://github.com/jdob/visitors-operator/blob/2a5adc5a08fc7a3fc182ef26e6da32640ab7119d/pkg/controller/visitorsapp/visitorsapp_controller.go#L115

**Comment by jdob** [comment]
It would work to check on err. I straight up hadn't considered the heap allocations, likely because I'm still pretty new to Go and don't quite have my head in that space yet. But this is an easy change I can do.

**Comment by embano1** [comment]
Yeah, don't worry too much on the heap (that's just a "side effect" here). But it would be more idiomatic to check the err and just return `reconcile.Result{}` and the `err` from the function.
